File: CSharp/[Udemy] CSharp POO/session9/Session90Ex2/Entities/Post.cs
using System.ComponentModel;
using System.Text;

namespace Session90Ex2.Entities
{
    internal class Post
    {
        public DateTime Moment { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public int Likes { get; set; }
        public List<Comment> comments { get; set; } = new List<Comment>();
        public Post()
        {
        }
        public Post(DateTime moment, string title, string content, int likes)
        {
            Moment = moment;
            Title = title;
            Content = content;
            Likes = likes;
        }

        public void AddComment(Comment comment)
        {
            comments.Add(comment);
        }
        public void RemoveComment(Comment comment)
        {
            comments.Remove(comment);
        }
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendLine(Title);
            sb.AppendLine($"{Likes} Likes - {Moment}");
            sb.AppendLine(Content);
            sb.AppendLine("comments:");
            foreach (Comment comment in comments)
            {
                sb.AppendLine(comment.Text);
            }
            return sb.ToString();
        }
    }
}

File: Java/[Udemy] Java POO/section12/StringBUILDER_ex/src/entities/Post.java
package entities;

import java.time.Instant;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;
import java.util.List;

public class Post {
	
	private Instant date;
	private String title;
	private String content;
	private Integer likes;
	
	private List<Comment> comments = new ArrayList<>();
	
	public Post(Instant date, String title, String content, int likes) {
		this.date = date;
        this.title = title;
        this.content = content;
        this.likes = likes;;
	}

	public Instant getDate() {
		return date;
	}

	public void setDate(Instant date) {
		this.date = date;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public Integer getLikes() {
		return likes;
	}

	public void setLikes(Integer likes) {
		this.likes = likes;
	}
	
	public void addComment(Comment coment) {
		comments.add(coment);
	}
	
	public void removeComment(Comment coment) {
        comments.remove(coment);
    }
	
	public List<Comment> getComments() {
        return comments;
    }
	
	@Override
    public String toString() {
		DateTimeFormatter fmt = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss").withZone(ZoneId.systemDefault());
        StringBuilder sb = new StringBuilder();
        sb.append(String.format("%s%n", getTitle()));
        sb.append(String.format("%d Likes - %s%n", getLikes(), fmt.format(getDate())));
        sb.append(String.format("%s%n%n", getContent()));
        sb.append(String.format("Comments:%n"));
        
        for (Comment coment : comments) {
            sb.append(String.format("%s%n", coment.getText()));
        }
        
        sb.append(String.format("----------------------------%n"));
        
        return sb.toString();
    }

}
